tamr_toolbox/utils/version.py

```diff
@@ -49,8 +53,128 @@ def _as_float(version: str) -> float:
     return (version_split[0] * 1000) + version_split[1] + (version_split[2] / 10)
 
 
+def _get_tamr_versions_from_function_args(*args, **kwargs) -> List[str]:
+    all_args = locals()
+    args = [arg for arg in all_args["args"]]
+    kwargs = list(all_args["kwargs"].values())
+    all_args_parsed = args + kwargs
+    response = []
+
+    # Return the client version (if we can find it)
+    for arg in all_args_parsed:
+        if type(arg) is Client:
+            response.append(current(arg))
+        elif hasattr(arg, "client"):
+            if type(arg.client) is Client:
+                response.append(current(arg.client))
+
+    return response
+
+
+def is_version_condition_met(
+    tamr_version: str,
+    min_version: str,
+    max_version: Optional[str] = None,
+    exact_version: bool = False,
+    raise_error: bool = False,
+) -> bool:
+    """
+    Check Tamr version (and raise error if appropriate).
+
+    Args:
+        tamr_version:
+            The version of Tamr being considered
+        min_version:
+            The minimum version of Tamr to test against
+        max_version:
+            The maximum version of Tamr to test against.
+            Default None, in which case no max version is tested for.
+        exact_version:
+            Compare against only one release of Tamr. Default is False
+        raise_error:
+            If True, raise an error if the version condition is not met. Default is False.
+
+    Raises:
+        ValueError: if `min_version` is greater than `max_version`
+        EnvironmentError: if `raise_error` is True, and the condition is not met
+
+    Notes:
+        If exact_version is True, max_version will be ignored
+
+    See Also:
+        utils.version.func_requires_tamr_version
+    """
+
+    error_str = None
+    if exact_version:
+        if not parse(min_version) == parse(tamr_version):
+            error_str = f"must be exactly {min_version}."
+
+    elif max_version:
+        if parse(min_version) > parse(max_version):
+            raise ValueError("min_version must be smaller than max_version")
+
+        if not parse(min_version) <= parse(tamr_version) <= parse(max_version):
+            error_str = f"must be between {min_version} and {max_version}."
+
+    elif not parse(min_version) <= parse(tamr_version):
+        error_str = f"but must be at least {min_version}."
```

Comment:
I think this ends up giving a repeated "but".
```suggestion
        error_str = f"must be at least {min_version}."
```